### ISSUE TITLE
add fish completion for pacaur

### ISSUE THREAD

**Issue by marmistrz** [fish-shell/fish-shell#2804]
Bash autocompletes the pacaur command (it's an AUR helper). It'd be great if fish did it too.

I can try making a pull request myself if you help me where to start.

**Comment by faho** [comment]
Of course. You'll probably want to look at the completions for related tools, like [pacman itself](https://github.com/fish-shell/fish-shell/blob/master/share/completions/pacman.fish) and [cower](https://github.com/fish-shell/fish-shell/blob/master/share/completions/cower.fish). You might be able to inherit pacman's completions via `complete -c pacaur -w pacman` - dunno if that makes sense, I rolled [my own stupid aur helper](https://github.com/faho/config/blob/master/fish/.config/fish/functions/aur.fish).

**Comment by marmistrz** [comment]
Ok, `complete -c ...` is a major improvement. But there's still something to be done. 

When `pacaur` searches for packages `pacaur -Ss packname`, it connects to the Internet to search for the packages in AUR. Still, it takes the pacman (official) packages from the local cache.

A sensible option would be to execute the `-Ss` command on each tab press - but is it a good idea? And what is the corresponding entry in `pacman.fish` - 'cause I can't find it.

**Comment by faho** [comment]
>And what is the corresponding entry in pacman.fish - 'cause I can't find it.

It's in a helper function in functions/__fish_print_packages.fish. (Because other commands might want packages too, though I don't think we have any distro-agnostic ones)

>A sensible option would be to execute the -Ss command on each tab press - but is it a good idea? 

It's possible to do some caching. The cower completions currently do something like it, but I'm not sure about it. It can be rather slow (I know, I'm on 2G some of the time) and the user might not expect it. If we had some way of streaming in new options as they appear (there's an issue open about that I think) it might be worth it, but as it stands I currently don't think so.

So you should rather add the additional options pacaur adds and just leave this for maybe later.

**Comment by marmistrz** [comment]
Hmm... Something weird happens. When I copy my newly created .fish file with completions to `/usr/share/fish/completions/`, then the wrapped completions are not recognized...

It's the whole file:
```
set -l progname pacaur

# complete -c $progname -s d -f -l download -d 'Clone the target(s) build files. Pass twice to fetch dependencies (done recursively).'
complete -c $progname -w pacman
```

**Comment by faho** [comment]
>When I copy my newly created .fish file with completions to /usr/share/fish/completions/, then the wrapped completions are not recognized...

FIrst of all, you'll want to keep local completions in ~/.config/fish/completions/.

Secondly: Which fish version are you using? We had some changes to the pacman completions (IIRC it used a function because it was written before the "-w" option to complete was added, this might interfere with actual wrappers).

This works for me.

**Comment by marmistrz** [comment]
2.2.0

**Comment by faho** [comment]
Okay, I just tried 2.2.0, and I believe the issue might be that in 2.2.0 you only get options if you have already entered a "-", ie. `pacman<TAB>` completes files (because no completion matches).

I hacked around this after 2.2.0.

Please try wrapping pacaur and then completing `pacaur -`. The rest should be fixed in the next release (real soon now).

**Comment by marmistrz** [comment]
Well, now after log in/out the thing works as expected (after entering the hyphen). But before the logout the only autocompletion which worked was the uncommented `-d`...

Do we need to close all fish instances for the new completions to get recognized?

**Comment by marmistrz** [comment]
Besides, I for the `-d` option, I copied the manpage description:

    Clone the target(s) build files. Pass twice to fetch dependencies (done recursively)

Isn't that too long?

**Comment by faho** [comment]
>Do we need to close all fish instances for the new completions to get recognized?

Usually no.

>Isn't that too long?

Yes - I'd make it something like "Clone build files. Pass twice for dependencies" or just "Clone build files".

**Comment by faho** [comment]
Fixed by #2934.
